### ISSUE TITLE
High local loss while finetuning resnet50

### ISSUE THREAD

**Issue by rishabhm12** [facebookresearch/VICRegL#14]
Hi Adrien,
I am trying to finetune resnet50 model using the complete checkpoint (excluding the classifier) on an e-commerce image dataset.
I am seeing very high variance, covariance & invariance term (100x) for local loss as compared to the logs you have shared in the repo. Also, the global loss terms are within the same range if I compare with it with your logs.

Additional Info: Since I am finetuning, I am using the same imagenet mean & std in transforms.

I wanted your opinion on why could this happen:
- There's something wrong with the local loss computation
-  Needs more # epochs to finetune ( I think won't help wrt local loss terms)
-  augmentations are very diff b/w pretrained model  you shared and the codebase

This is the command line args passed to start finetuning:
`VICRegL main_vicregl.py --local_rank=0 --fp16 --exp-dir ./resnet_finetuning/ --arch resnet50 --epochs 30 --batch-size 512 --optimizer adamw --base-lr 0.00075 --weight-decay 1e-06 --size-crops 224 --num-crops 2 --min_scale_crops 0.08 --max_scale_crops 1.0 --alpha 0.75`

**Repos logs for resnet50_075alpha at 30th epoch and 300th epoch:**
`{"ep": 29, "st": 18732, "lr": 1.5814172938771736, "t": "000017683", "stdr": 0.0154, "stde": 0.042409, "corr": 0.008171, "core": 0.000214, "inv_l": 3.825874, "std_l": 9.492188, "cov_l": 2.068358, "minv_l": 0.213284, "mvar_l": 0.053986, "mcov_l": 0.137415, "cls_l": 2.682755, "top1": 42.2363, "top5": 67.4316, "l": 22.093506}`

`{"ep": 299, "st": 187665, "lr": 0.0016000029958416245, "t": "000176214", "stdr": 0.016162, "stde": 0.007431, "corr": 0.008438, "core": 0.000113, "inv_l": 3.110444, "std_l": 3.189453, "cov_l": 3.890673, "minv_l": 0.07154, "mvar_l": 0.042328, "mcov_l": 0.203255, "cls_l": 1.158083, "top1": 72.6074, "top5": 89.1113, "l": 14.522144}`

**My finetuning log at 30th epoch:**
`{"ep": 29, "st": 26650, "lr": 1.5092537293760234e-06, "t": "000031628", "stdr": 0.01549, "stde": 0.011048, "corr": 0.00642, "core": 1e-06, "inv_l": 1.40463, "var_l": 9.71875, "cov_l": 3.121696, "total_loss": 15.580208, "minv_l": 2.408733, "mvar_l": 8.0, "mcov_l": 9.176867}`

I am concerned about **minv_l, mvar_l and mcov_l**  terms

Looking forward to know your thoughts.
Thank you!

**Comment by rishabhm12** [comment]
In the paper it's mentioned 

> No color augmentation are used for either training
and evaluation

But the code has it 
https://github.com/facebookresearch/VICRegL/blob/1bb35167bd8754d2ccb80b599e6592ea60976fb2/transforms.py#L305

**Comment by rishabhm12** [comment]
In the paper it's mentioned, top y1 and y2 pairs are used which are [20,4] respectively but in the code I don't think we are using this filtering. 
https://github.com/facebookresearch/VICRegL/blob/1bb35167bd8754d2ccb80b599e6592ea60976fb2/main_vicregl.py#L62
https://github.com/facebookresearch/VICRegL/blob/1bb35167bd8754d2ccb80b599e6592ea60976fb2/main_vicregl.py#L321

**Comment by Adrien987k** [comment]
Hi,

About the loss that is significantly greater than our log file, that is related to a fix that we made to the code recently, where the loss was not correctly aggregated see:
https://github.com/facebookresearch/VICRegL/issues/6
There is nothing to worry about this is normal.

About the data augmentation, this sentence refers to the linear evaluation training and evaluation phases, which are different from the pretraing phase. Basically, you pretrain, then you train a linear classifier on top of the frozen features, then you evaluate your classifier.

About the filtering, we found after that only filtering for the location-based matching works great. This should not make any difference.

Let me know if you still have questions or issue.
